### DISCUSSION TITLE
Is there a tutorial for writing goldmark extensions?

### DISCUSSION THREAD

**Discussion by j2kun** [yuin/goldmark#436] [Q&A]
I see lots of examples but I am looking for an explanation of what is needed to implement a basic goldmark extension, such as a parser extension.

**Comment by maaz-munir** [comment]
yea something like a guide would be really useful!

**Comment by j2kun** [comment]
I ended up having to teach myself while writing this one: https://github.com/j2kun/hugo-goldmark-extensions/tree/main/passthrough

Unfortunately I had to implement it a few times because my initial attempts didn't work properly with goldmark's internals
